app/controllers/application_controller.rb

```diff
@@ -1,10 +1,142 @@
-require './config/environment'
-
 class ApplicationController < Sinatra::Base
+set :views, Proc.new { File.join(root, "../views/") }
 
   configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "my_little_secret"
+  end
+
+    get '/' do
+
+      erb :index
+    end
+
+#------- Tweet Controls ------------#
+    get '/tweets' do
+      if Helpers.logged_in?(session)
+        @user = User.find(session[:user_id])
+        @tweets = Tweet.all
+         erb :'/tweets/tweets'
+      else
+        redirect '/users/login'
+      end
+   end
+
+   get '/tweets/new' do
+     if Helpers.logged_in?(session)
+       @user = User.find(session[:user_id])
+       erb :'/tweets/create_tweet'
+     else
+       redirect '/users/login'
+     end
+   end
+
+   get "/tweets/:id/edit" do
+     if Helpers.logged_in?(session) && Helpers.current_user(session).id == session[:user_id]
+       @tweet = Tweet.find(params[:id])
+       erb :'/tweets/edit_tweet'
+     else
+       redirect '/login'
+     end
+   end
+
+   get "/tweets/:id" do
+     if Helpers.logged_in?(session)
+       @tweet = Tweet.find(params[:id])
+       erb :'/tweets/show_tweet'
+     else
+       redirect '/login'
+     end
+   end
+
+
+   patch '/tweets/:id' do
+      @tweet = Tweet.find(params[:id])
+     if !params[:content].empty?
+       @user = @tweet.user
+       @tweet.content  = params[:content]
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...